reentrancy-counter/src/agent.spec.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  Finding,
+  HandleTransaction,
+  createTransactionEvent,
+  Trace,
+  Transaction,
+  Receipt,
+  Block,
+  TransactionEvent,
+  TraceAction,
+} from "forta-agent";
+import agent from "./agent";
+import { createFinding, reentracyLevel } from "./agent.utils";
+
+const createTrace = (to: string, stack: number[]): Trace => {
+  return {
+    traceAddress: stack,
+    action: {
+      to: to,
+    } as TraceAction,
+  } as Trace;
+};
+
+const createTxEvent = (traces: Trace[]) =>
+  createTransactionEvent({
+    transaction: {} as Transaction,
+    receipt: {} as Receipt,
+    block: {} as Block,
+    traces: traces,
+  });
+
+
+describe("Reentrancy counter agent tests suit", () => {
+  const handleTransaction: HandleTransaction = agent.handleTransaction;
+
+  describe("handleTransaction", () => {
+    it("Should return empty findinds if no traces provided", async () => {
+      const tx: TransactionEvent = createTxEvent([]);
+      const findings: Finding[] = await handleTransaction(tx);
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("Should return empty findinds if no repetition detected", async () => {
+      const tx: TransactionEvent = createTxEvent([
+        createTrace("0x0", []),
+        createTrace("0x1", [0]),
+        createTrace("0x2", [0, 0]),
+        createTrace("0x3", [0, 0, 0]),
+      ]);
+      const findings: Finding[] = await handleTransaction(tx);
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("Should ignore non reentrant calls", async () => {
+      const tx: TransactionEvent = createTxEvent([
+        createTrace("0x0", []),
+        createTrace("0x1", [0]),
```

Comment:
More description on what 0, 1 ... represents ( array elements ) would be nice to have in comments

Reply:
as you can see in the `createTrace` function the array represent the `traceAddress` field of a `Trace` that is why no explanation was added there, but in short the array represent the sequence of calls made that originated the `Trace` and the number represent the depth of each call.
You can read about it here https://openethereum.github.io/JSONRPC-trace-module